### PR TITLE
👾 Update Dependencies

### PR DESCRIPTION
**Pull request by demariadaniel**
## Link to Issue

Relates to Dependabot PRs: https://github.com/icgc-argo/argo-clinical/pulls/app%2Fdependabot

## Description

Fixes Vulnerabilities:
- Develop branch: 2 low, 21 moderate, 18 high, 2 critical
- This PR: 1 moderate, 1 high

Note: I've created a separate feature branch for fixing the integration tests we can use if anyone is hesitant to merge this to Develop

### Type of Change

- [x] Bug
- [ ] Refactor
- [ ] New Feature
- [ ] Release Candidate

### Checklist before requesting review:

- [ ] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

**Comment by ciaranschutte** [comment]
I see it's still a WIP but I'm browsing by "open" PRs - just to note anything with a major version will need to be well tested. eg. `nodemon` https://github.com/icgc-argo/argo-clinical/blob/develop/package.json#L23-L24
major version bump = breaking changes and it's good to verify none of those changes affect us. 
most of the libs should have a list of breaking changes that could affect us.

**Comment by demariadaniel** [comment]
> I see it's still a WIP but I'm browsing by "open" PRs - just to note anything with a major version will need to be well tested. eg. `nodemon` https://github.com/icgc-argo/argo-clinical/blob/develop/package.json#L23-L24 major version bump = breaking changes and it's good to verify none of those changes affect us. most of the libs should have a list of breaking changes that could affect us.


There is a follow up WIP PR as well to fix the integration tests (based off this branch). I'll try to get that done first so we have some additional validation that these updates will work.

**Comment by demariadaniel** [comment]
Created feature branch `feature/fix-integration-tests` so all changes can be thoroughly tested in isolation before merging

File: package.json
Comment (demariadaniel) [inline package.json:88]: [Changelog](https://github.com/remy/nodemon/releases)

Comment (demariadaniel) [reply package.json:88]: Breaking changes don't look like an issue, support for Node 8 is dropped: 
https://github.com/remy/nodemon/releases/tag/v3.0.0

File: package.json
Comment (demariadaniel) [inline package.json:100]: Minor version change, bug fixes, additional options + fields: https://github.com/apollographql/apollo-server/blob/main/packages/server/CHANGELOG.md

File: package.json
Comment (demariadaniel) [inline package.json:123]: Breaking changes: https://github.com/auth0/node-jsonwebtoken/blob/master/CHANGELOG.md

Potential impact? Will have to assess. https://github.com/icgc-argo/argo-clinical/blob/develop/src/decorators/index.ts#L55

**Review by ciaranschutte — APPROVED** [review]
updates look ok.

approving so you can merge after assessment https://github.com/Pan-Canadian-Genome-Library/daco/pull/7#discussion_r1809974318

File: package.json
Comment (demariadaniel) [reply package.json:123]: So breaking changes:
- No longer supports Node 11 (not an issue)
- No longer supports unsigned tokens by default (shouldn't be an issue)
- RSA key must be >2048 bits, we are using RS256 so I believe this is fine

**Pull request merged**
